client/my-sites/email/email-management/list/index.jsx

```diff
@@ -0,0 +1,243 @@
+
+const SITE_CONTEXT_NO_DOMAIN_NO_PLAN = 'no_domain_no_plan';
+/* TODO: const SITE_CONTEXT_NO_DOMAIN_WITH_PLAN = 'no_domain_with_plan'; */
+
+const USER_ACTION_JUST_DOMAIN = 'just_domain';
+const USER_ACTION_START_UPGRADE = 'start_upgrade';
+
+class EmailManagementList extends React.Component {
+	static propTypes = {
+		currentRoute: PropTypes.string.isRequired,
+		domains: PropTypes.array.isRequired,
+		selectedSite: PropTypes.object.isRequired,
+		selectedSiteId: PropTypes.number.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	handleJustDomainClick = ( siteContext ) => {
+		this.recordCtaClick( siteContext, USER_ACTION_JUST_DOMAIN );
+
+		page( domainAddNew( this.props.selectedSite.slug ) );
+	};
+
+	handleUpgradeClick = ( siteContext ) => {
+		this.recordCtaClick( siteContext, USER_ACTION_START_UPGRADE );
+	};
+
+	recordCtaClick( siteContext, userAction ) {
+		recordTracksEvent( 'calypso_email_management_no_domain_cta_click', {
+			user_action: userAction,
+			site_context: siteContext,
+		} );
+	}
+
+	handleAddEmailToDomainClick = ( domainName ) => {
+		const { currentRoute, selectedSite } = this.props;
+
+		recordTracksEvent( 'calypso_email_management_add_email_to_domain', { domain: domainName } );
+
+		page( emailManagement( selectedSite.slug, domainName, currentRoute ) );
+	};
+
+	renderNoDomainNoPlanCta() {
+		const { translate } = this.props;
+
+		const upgradeButton = (
+			<Button primary onClick={ () => this.handleUpgradeClick( SITE_CONTEXT_NO_DOMAIN_NO_PLAN ) }>
+				{ translate( 'Upgrade and claim my domain' ) }
+			</Button>
+		);
+
+		const justDomainButton = (
+			<Button onClick={ () => this.handleJustDomainClick( SITE_CONTEXT_NO_DOMAIN_NO_PLAN ) }>
+				{ translate( 'Just buy a domain' ) }
+			</Button>
+		);
+
+		return (
+			<Card>
+				<EmptyContent
+					action={ upgradeButton }
+					illustration="calypso/assets/images/customer-home/illustration--task-find-domain.svg"
+					line={ translate(
+						'Upgrade now and claim your domain (or just buy a domain!) and pick from one of our flexible options ' +
+							'to connect your domain with email and start getting emails at your custom email address today.'
+					) }
+					secondaryAction={ justDomainButton }
+					title={ translate( 'Get your own domain for a custom email address' ) }
+				/>
+			</Card>
+		);
+	}
+
+	renderDomainsWithActiveEmail( domainsWithActiveEmail ) {
+		if ( domainsWithActiveEmail.length === 0 ) {
+			return null;
+		}
+
+		const { selectedSite, translate } = this.props;
+
+		const activeEmailItems = domainsWithActiveEmail.map( ( domain, index ) => {
+			return (
+				<EmailManagementListItem
+					domain={ domain }
+					key={ `email-domain-${ domain.name }--${ index }` }
+					selectedSite={ selectedSite }
+				/>
+			);
+		} );
+
+		return (
+			<React.Fragment>
+				<SectionHeader label={ translate( 'Active Email Plans' ) } />
+				<div className="email-management-list__active-emails">{ activeEmailItems }</div>
+			</React.Fragment>
```

Comment:
Note you can also use the `<>[...]</>` shorter notation.

Reply:
TIL! I'll take that on today.